static/js/notes.js

```diff
@@ -38,7 +100,30 @@ const $createNote = function(note) {
         .append($content);
 };
 
+// ADD NOTE
+const $createAddNoteModal = function() {
+    const $modal = $('<div>').addClass('modal');
+
+    const $buttons = NOTE_TYPES.map(noteType => {
+        return $createButtonWithText(noteType).addClass(classNamesByName.noteTypeButton);
+    });
+
+    const $textarea = $createTextarea('Note').addClass('note-content');
+
+    return $modal
+        .append($createGridContainer()
+            .append($createRowWithTwoElements($buttons[0], $buttons[1]))
+            .append($createRowWithTwoElements($buttons[2], $buttons[3])))
+        .append($textarea);
+};
+
+const activateModal = function($modal) {
+    mui.overlay('on', $modal.get(0));
+};
+
 $(function() {
+    const $addNoteModal = $createAddNoteModal();
```

Comment:
I was pretty confused by this for a while, i was like its a function that adds a note modal. But i was like its seems like its a static object, its not a function that will add a note modal. 

Then it finally hit me its a modal to add notes. *facepalm*